packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -41,23 +49,28 @@ export const TimePicker: React.FunctionComponent<ITimePickerProps> = ({
 
   const optionsCount = getDropdownOptionsCount(increments, timeRange);
 
+  const baseDate: Date = React.useMemo(() => generatebaseDate(increments, timeRange, defaultValue), [
```

Comment:
Per the standard pattern of controlled and uncontrolled components, `defaultValue` is only for if the component is uncontrolled, so updates to `defaultValue` shouldn't be respected--either directly as the value, or in other places within the component. In this case, probably the right thing to do is store the first `defaultValue` passed in and use that when calculating the base date. 
```ts
// top of file
import { useConst } from '@fluentui/react-hooks';
// here
const initialValue = useConst(defaultValue || new Date());
const baseDate = React.useMemo(() => generateBaseDate(increments, timeRange, initialValue), [increments, timeRange, initialValue]);
```

Reply:
Resolved by storing defaultValue via suggested code snippet.